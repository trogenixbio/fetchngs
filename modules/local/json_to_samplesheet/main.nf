process JSON_TO_SAMPLESHEET {
    tag "$sample_json"
    label 'error_retry'

    conda "conda-forge::python=3.9.5"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    path sample_json
    val nf_core_pipeline

    output:
    path "samplesheet.csv"      , emit: samplesheet
    path "versions.yml"         , emit: versions

    script:
    def pipeline = "${nf_core_pipeline}" ?: ''
    def include  = ''
    def exclude  = 'strandedness,replicate,fasta'

    if (pipeline) {
        if (pipeline == 'rnaseq') {
            include = "strandedness"
            exclude = "replicate,fasta"
        } else if (pipeline == 'atacseq') {
            include = "replicate"
            exclude = "strandedness,fasta"
        } else if (pipeline == 'taxprofiler') {
            include = "fasta"
            exclude = "replicate,strandedness"
        }
    }

    """
    json_to_samplesheet.py \\
        ${sample_json} \\
        samplesheet.csv \\
        --include ${include} \\
        --exclude ${exclude}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version | sed 's/Python //g')
    END_VERSIONS
    """
}
