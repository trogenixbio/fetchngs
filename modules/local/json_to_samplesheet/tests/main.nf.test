nextflow_process {

    name "Test Process JSON_TO_SAMPLESHEET"
    script "../main.nf"
    process "JSON_TO_SAMPLESHEET"

    test("Should run without failures - empty") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[1] = ""
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - rnaseq") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[1] = "rnaseq"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - atacseq") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[1] = "atacseq"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("Should run without failures - taxprofiler") {

        when {
            process {
                """
                input[0] = file("$baseDir" + "/" + params.modules_testdata_base_path_local + "samplesheet.json", checkIfExists: true)
                input[1] = "taxprofiler"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }
}
