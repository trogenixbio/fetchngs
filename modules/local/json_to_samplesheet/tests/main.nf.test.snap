{
    "Should run without failures - atacseq": {
        "content": [
            {
                "0": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "1": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "versions": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T13:37:13.959099735"
    },
    "Should run without failures - rnaseq": {
        "content": [
            {
                "0": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "1": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "versions": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T13:37:04.603093493"
    },
    "Should run without failures - taxprofiler": {
        "content": [
            {
                "0": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "1": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "versions": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T13:37:23.121683771"
    },
    "Should run without failures - empty": {
        "content": [
            {
                "0": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "1": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ],
                "samplesheet": [
                    "samplesheet.csv:md5,92bcb32506bd1c8be02b0767420fbf8c"
                ],
                "versions": [
                    "versions.yml:md5,44e7d59dcc70c98c584aaa78bf3a166f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T13:36:54.896842335"
    }
}